File: DiazNayely-R3-U2/DiazNayely-R3-U2/ViewController.m
//
//  ViewController.m
//  DiazNayely-R3-U2
//
//  Created by Ismael Sánchez Rios on 11/12/23.
//

#import "ViewController.h"
#import "ViewController2.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Configurar la imagen de bienvenida al usuario
    UIImage *welcomeImage = [UIImage imageNamed:@"pantallabienvenida"];
    self.welcomeImageView = [[UIImageView alloc] initWithImage:welcomeImage];
    self.welcomeImageView.contentMode = UIViewContentModeScaleAspectFill; // Ajustar el modo de contenido según sea necesario
    self.welcomeImageView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:self.welcomeImageView];
    
    // Configurar el botón de comenzar
    self.startButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [self.startButton setTitle:@"Comenzar" forState:UIControlStateNormal];
    [self.startButton addTarget:self action:@selector(startButtonTapped:) forControlEvents:UIControlEventTouchUpInside];
    self.startButton.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:self.startButton];
    
    // Configurar autolayout para posicionar elementos
    [NSLayoutConstraint activateConstraints:@[
        // Imagen de bienvenida ocupa toda la pantalla
        [self.welcomeImageView.topAnchor constraintEqualToAnchor:self.view.topAnchor],
        [self.welcomeImageView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.welcomeImageView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.welcomeImageView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor],
        
        // Botón en la parte inferior
        [self.startButton.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
        [self.startButton.widthAnchor constraintEqualToConstant:120.0], // Ajustar el ancho según sea necesario
        [self.startButton.heightAnchor constraintEqualToConstant:40.0], // Ajustar el alto según sea necesario
        [self.startButton.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:-40.0],
    ]];
}

- (void)startButtonTapped:(UIButton *)sender {
    // Acción al presionar el botón "Comenzar"
    
    // Crear e iniciar la instancia del segundo controlador de vista (ViewController2)
    ViewController2 *secondViewController = [[ViewController2 alloc] initWithNibName:@"ViewController2" bundle:nil];
    
    // Presentar el segundo controlador de vista
    [self presentViewController:secondViewController animated:YES completion:nil];
}




@end

File: DiazNayelyR5U3/DiazNayelyR5U3/ViewController.swift
import UIKit

class ViewController: UIViewController {

    let imageView: UIImageView = {
        let imageView = UIImageView()
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.contentMode = .scaleAspectFill // Escoge el modo de llenado que prefieras
        imageView.image = UIImage(named: "pantallainicio") 
        return imageView
    }()

    let startButton: UIButton = {
        let button = UIButton()
        button.translatesAutoresizingMaskIntoConstraints = false
        button.setTitle("Comenzar", for: .normal)
        button.setTitleColor(.blue, for: .normal)
        button.addTarget(self, action: #selector(startButtonTapped), for: .touchUpInside)
        return button
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white

        // Agregar la imagen de bienvenida al controlador
        view.addSubview(imageView)
        NSLayoutConstraint.activate([
            imageView.topAnchor.constraint(equalTo: view.topAnchor),
            imageView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            imageView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
            imageView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
        ])

        // Agregar el botón "Comenzar" al controlador
        view.addSubview(startButton)
        NSLayoutConstraint.activate([
            startButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
            startButton.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -50) // Ajusta la distancia del botón al borde inferior
        ])
    }

    @objc func startButtonTapped() {
        print("start")
        let questionViewController = QuestionViewController()
        questionViewController.modalPresentationStyle = .fullScreen
        present(questionViewController, animated: true, completion: nil)
    }
}

